flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SymbolArgumentTypeStrategy.java

```diff
@@ -45,7 +44,8 @@ public SymbolArgumentTypeStrategy(Class<? extends Enum<? extends TableSymbol>> s
     public Optional<DataType> inferArgumentType(
             CallContext callContext, int argumentPos, boolean throwOnFailure) {
         final DataType argumentType = callContext.getArgumentDataTypes().get(argumentPos);
-        if (argumentType.getLogicalType().getTypeRoot() != LogicalTypeRoot.SYMBOL) {
+        if (argumentType.getLogicalType().getTypeRoot() != LogicalTypeRoot.SYMBOL
+                || !callContext.isArgumentLiteral(argumentPos)) {
```

Comment:
Why is this needed now? is it tested somewhere?

Reply:
This is always needed before calling `getArgumentValue` as written in the JavaDocs. A user would need to create a weird expression to have a symbol that is not a literal.